extensions/wikia/PageShare/tests/PageShareHelperTest.php

```diff
@@ -128,29 +111,27 @@ public function testGetLangForPageShare(
 			->will( $this->returnValue( $isAnon ) );
 		$this->mockGlobalVariable( 'wgUser', $wgUserMock );
 
-		$this->assertEquals( $expectedResult, PageShareHelper::getLangForPageShare() );
+		$this->assertEquals( $expectedResult, PageShareHelper::getLangForPageShare($browserLang, $requestLang) );
 	}
 
 	/**
 	 * Data provider for testGetLangForPageShare.
 	 *
-	 * First set - anon user, language should be taken from HTTP_ACCEPT_LANGUAGE header
-	 * Second set - anon user, HTTP_ACCEPT_LANGUAGE header is not set, return default language defined in SHARE_DEFAULT_LANGUAGE
-	 * Third set - anon user, HTTP_ACCEPT_LANGUAGE header is set but is "falsy", return default language defined in SHARE_DEFAULT_LANGUAGE
-	 * Fourth set - anon user, language is overwritten by ?uselang=xx
-	 * Fifth set - logged in user, language is taken from user language
-	 * Sixth set - logged in user, language is overwritten by ?uselang=xx
+	 * First set - anon user, language should be taken from browser language
+	 * Second set - anon user, browser language is set to "falsy" value, return default language defined in SHARE_DEFAULT_LANGUAGE
+	 * Third set - anon user, language is overwritten by ?uselang=xx
+	 * Fourth set - logged in user, language is taken from user language
+	 * Fifth set - logged in user, language is overwritten by ?uselang=xx
 	 *
 	 * @return array
 	 */
 	public function getLangForPageShareDataProvider() {
 		return [
-			[ null, null, true, true, 'pl,pl;q=0.8', 'pl' ],
-			[ null, null, true, false, 'pt-br,pt;q=0.8', 'en' ],
-			[ null, null, true, true, false, 'en' ],
-			[ 'bar', null, true, true, 'pt-br,pt;q=0.8', 'bar' ],
-			[ null, 'foo', false, false, null, 'foo' ],
-			[ 'bar', 'foo', false, false, null, 'bar' ]
+			[ null, null, true, 'pl', 'pl' ],
+			[ null, null, true, false, 'en' ],
+			[ 'bar', null, true, 'pt-br,pt;q=0.8', 'bar' ],
```

Comment:
The main goal of this test is to check if "bar" is overwriting value of "pt..." so the value of browser lang argument is irrelevant. I can change it to "foo" if it increases readability.